src/conversions/chrono.rs

```diff
@@ -262,7 +262,7 @@ impl FromPyObject<'_> for DateTime<FixedOffset> {
                 .ok_or_else(|| PyValueError::new_err("invalid or out-of-range time"))?,
         );
         // `FixedOffset` cannot have ambiguities so we don't have to worry about DST folds and such
-        Ok(DateTime::from_local(dt, tz))
+        Ok(dt.and_local_timezone(tz).unwrap())
```

Comment:
I think we would need to bump our dependency on `chrono` to 0.4.20 if we use `and_local_timezone`.

Reply:
Good catch, I'll adjust the newsfragment accordingly.

I wonder if we should put this minimum version into CI (maybe for the MSRV job). E.g. https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html

Reply:
I think minimum versions do not require any special handling and in any case our lockfile does not affect downstream users, so I think a simple `chrono = { version = "0.4.20", optional = true }` is correct and sufficient.

Reply:
So we actually need 0.4.25 for `and_utc`, I'll increase to that.